bot/plotting/comparison_generator.py

```diff
@@ -46,96 +98,126 @@ def comparison_generator(
     while True:
         try:
 
-            # generate a list of parameter values by varying a single parameter
-            # if only 1 model is selected
+            # if testing, don't select simulations randomly
+            if provided_choice is not None:
+                choice = provided_choice
+            else:
+                choice_list = ["experiment", "no experiment"]
+                choice = random.choice(choice_list)
+
+            # remove "Current function [A]" from the dict if simulating an
+            # experiment and add it back if not an experiment
+            # (adding it back because pop edits the original list)
```

Comment:
how come you need to add it back? if it's not an experiment you don't enter the branch where you do `pop`?

Reply:
The problem comes in when there is an error while solving. If the first choice was an `"experiment"` and `"Current function [A]"` was popped out, but there was an error while solving, the loop will iterate (keeps on iterating until there is no error) and might select `"no experiment"` as the choice this time, but now `"Current function [A]"` would be gone as `pop` edited the original `dict`.